contracts/SafeProtocolManager.sol

```diff
@@ -286,4 +290,117 @@ contract SafeProtocolManager is ISafeProtocolManager, RegistryManager, HooksMana
+     * @param data Artibtrary length bytes containing payload
+     * @param operation Call or DelegateCall operation
+     * @param safeTxGas uint256
+     * @param baseGas uint256
+     * @param gasPrice uint256
+     * @param gasToken address
+     * @param refundReceiver payable address
+     * @param signatures Arbitrary bytes containing ECDSA signatures
+     * @param msgSender Sender of the transaction
+     */
+    function checkTransaction(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address payable refundReceiver,
+        bytes memory signatures,
+        address msgSender
+    ) external {
+        // Store hooks address in tempHooksAddress so that checkAfterExecution(...) and checkModuleTransaction(...) can access it.
+        address tempHooksAddressForSafe = tempHooksAddress[msg.sender] = enabledHooks[msg.sender];
+
+        if (tempHooksAddressForSafe == address(0)) return;
+        bytes memory executionMetadata = abi.encode(
+            to,
+            value,
+            data,
+            safeTxGas,
+            baseGas,
+            gasPrice,
+            gasToken,
+            refundReceiver,
+            signatures,
+            msgSender
+        );
+
+        if (operation == Enum.Operation.Call) {
+            SafeProtocolAction[] memory actions = new SafeProtocolAction[](1);
+            actions[0] = SafeProtocolAction(payable(to), value, data);
+            SafeTransaction memory safeTx = SafeTransaction(actions, 0, "");
+            ISafeProtocolHooks(tempHooksAddressForSafe).preCheck(ISafe(msg.sender), safeTx, 0, executionMetadata);
+        } else {
+            // Using else instead of "else if(operation == Enum.Operation.DelegateCall)" to reduce gas usage
+            // and Safe allows only Call and DelegateCall operations.
+            SafeProtocolAction memory action = SafeProtocolAction(payable(to), value, data);
+            SafeRootAccess memory safeTx = SafeRootAccess(action, 0, "");
+            ISafeProtocolHooks(tempHooksAddressForSafe).preCheckRootAccess(ISafe(msg.sender), safeTx, 0, executionMetadata);
+        }
+    }
+
+    /**
+     * @notice Implement BaseGuard interface to allow Safe to add Manager as a guard for existing Safe accounts (upto version 1.5.x).
+     * @param success bool
+     */
+    function checkAfterExecution(bytes32, bool success) external {
+        address tempHooksAddressForSafe = tempHooksAddress[msg.sender];
+        if (tempHooksAddressForSafe == address(0)) return;
+
+        // Use tempHooksAddress to avoid a case where hooks get updated in the middle of a transaction.
+        ISafeProtocolHooks(tempHooksAddressForSafe).postCheck(ISafe(msg.sender), success, "");
+
+        // Reset to address(0) so that there is no unattended storage
+        tempHooksAddress[msg.sender] = address(0);
```

Comment:
Added after post check because it is possible (highly not likely) that a hook would initiate a call through a safe that would call `checkTransaction` and leave storage uncleared after the transaction.

So, enforce that storage is cleared, it cleared after postcheck call

Reply:
I think it would depend what flow we expect, but no blocker